android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -1422,4 +1426,13 @@ public static boolean isBiometricAvailable(Context context) {
                     && fingerprintManager.hasEnrolledFingerprints();
         }
     }
+
+    @Nullable
+    public static Profile getProfile(boolean isIncognito) {
+        ChromeActivity chromeActivity = BraveActivity.getBraveActivity();
+        if (chromeActivity == null) chromeActivity = BraveActivity.getChromeTabbedActivity();
+        if (chromeActivity == null) return null;
```

Comment:
I put the conditions there just in case, but it shouldn't happen. That's why I have `assert profile != null` after the calls. It is different from tabs because this is Activity itself, even there is no tab, the `TabSelector` is still there and the only way Activity doesn't exist is either CustomTab or WebView if I'm not mistaken.

Reply:
let's add a check in all cc files of factory creation, to check if the passed profile is null to fallback to the previous way `auto* profile = ProfileManager::GetActiveUserProfile();`. It's hard to say right now what will happen with CustomTabs when we finish DApps integration, we will better remove it if we see not needed instead of seeing crashes.

Reply:
Ok I see. However I will add `Profile.getLastUsedRegularProfile()` here instead of adding in `cc` as Brian said redirecting profile in factory is a logical layer violation.

Reply:
Updated.